src/extension.ts

```diff
@@ -376,3 +398,185 @@ function deleteCustomResource(command: string) {
+		//There is no way to check if the yaml extension has been loaded
+		//So the only way to wait for it to load is to keep calling this
+		//command until it succeeds.
+		let docSymbols = undefined;
+		while(!docSymbols){
+			docSymbols = await vscode.commands.executeCommand(
+				'vscode.executeDocumentSymbolProvider',
+				document.uri
+			) as vscode.DocumentSymbol[];
+			//[Optional] Sleep to be mindful and not overload the command queue
+			if(!docSymbols){
+				await util.sleep(100);
+			}
+		}
+
+		//If we succesfuly read the galaxy data we proceed to lint those features
+		if(galaxyData !== undefined){
+			const galaxyConfig = yaml.load(galaxyData) as AnsibleGalaxyYmlSchema;
+
+			//Validate that operatorConfig values name, version, and domain match galaxy name, version, and namespace
+			if((galaxyConfig.namespace && operatorConfig.domain) && galaxyConfig.namespace !== operatorConfig.domain){
+				//Get domain symbol
+				const domainSymbol : vscode.DocumentSymbol | undefined = docSymbols.find( (symbol: vscode.DocumentSymbol) => (symbol.name === 'domain' && symbol.detail === operatorConfig.domain));
+				if(domainSymbol){
+					diagnostics.push({
+						range: domainSymbol.range,
+						message: 'Domain should match the namespace value specified in your galaxy.yml file, unless a fork/clone of an official Ansible Collection is desired.',
+						severity: vscode.DiagnosticSeverity.Warning,
+					});
+				}
+			}
+			if((galaxyConfig.name && operatorConfig.name) && galaxyConfig.name !== operatorConfig.name){
+				//Get name symbol
+				const nameSymbol : vscode.DocumentSymbol | undefined = docSymbols.find( (symbol: vscode.DocumentSymbol) => (symbol.name === 'name' && symbol.detail === operatorConfig.name));
+				if(nameSymbol){
+					diagnostics.push({
+						range: nameSymbol.range,
+						message: 'Name should match the name specified in your galaxy.yml file, unless a fork/clone of an official Ansible Collection is desired.',
+						severity: vscode.DiagnosticSeverity.Warning,
+					});
+				}
+			}
+			if((galaxyConfig.version && operatorConfig.version) && galaxyConfig.version !== operatorConfig.version){
+				//Get version symbol
+				const versionSymbol : vscode.DocumentSymbol | undefined = docSymbols.find( (symbol: vscode.DocumentSymbol) => (symbol.name === 'version' && symbol.detail === operatorConfig.version));
+				if(versionSymbol){
+					diagnostics.push({
+						range: versionSymbol.range,
+						message: 'Version should match the version specified in your galaxy.yml file.',
+						severity: vscode.DiagnosticSeverity.Error,
+					});
+				}
+			}
+		}
+
+		//validate that an ansible config file does not exist
+		try {
+			fs.readFileSync(path.join(path.dirname(document.uri.fsPath), 'ansible.cfg'), 'utf8');
+			diagnostics.push({
+				range: new vscode.Range(document.positionAt(0),document.positionAt(0)),
+				message: "Collection Must not contain an ansible.cfg file. Please delete it and reimport the collection",
```

Comment:
Can we allow this rule as long as this file is added to the `build_ignore` in the `galaxy.yml` file? As long as this file isn't in the final build, then this should be fine. 

And if this is allowed, then we should also update this message to say `"Collection build Must not contain an ansible.cfg file. Please delete it or add this file to the build_ignore section of the galaxy.yml file"`